testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -1250,20 +1340,55 @@ impl BlockMinerThread {
             }
         };
 
+        debug!(
+            "make_tenure_start_info: reason = {:?}, burn_view = {:?}, tenure_change_tx = {:?}",
+            &self.reason, &self.burn_block.consensus_hash, &tenure_change_tx
+        );
+
         Ok(NakamotoTenureInfo {
             coinbase_tx,
             tenure_change_tx,
         })
     }
 
+    /// Get the ongoing burn view in the chain state
+    pub fn get_ongoing_tenure_id(
+        sortdb: &SortitionDB,
+        chain_state: &mut StacksChainState,
+    ) -> Result<NakamotoTenureEventId, NakamotoNodeError> {
+        let cur_stacks_tip_header =
+            NakamotoChainState::get_canonical_block_header(chain_state.db(), sortdb)?
+                .ok_or_else(|| NakamotoNodeError::UnexpectedChainState)?;
+
+        let cur_stacks_tip_id = cur_stacks_tip_header.index_block_hash();
+        let ongoing_tenure_id = if let Some(tenure_id) = chain_state
+            .index_conn()
+            .get_ongoing_tenure_id(&cur_stacks_tip_id)?
+        {
+            // ongoing tenure is a Nakamoto tenure
+            tenure_id
+        } else {
+            // ongoing tenure is an epoch 2.x tenure, so it's the same as the canonical stacks 2.x
+            // tip
+            NakamotoTenureEventId {
+                burn_view_consensus_hash: cur_stacks_tip_header.consensus_hash,
+                block_id: cur_stacks_tip_id,
+            }
+        };
+        Ok(ongoing_tenure_id)
+    }
```

Comment:
I think this function is no longer used, so it can be removed.

Reply:
Sure